examples/cpp/hello_world/CLIParser.hpp

```diff
@@ -0,0 +1,172 @@
+
+#include <cstdlib>
+#include <iostream>
+
+#include <fastdds/dds/log/Log.hpp>
+
+#ifndef _FASTDDS_HELLO_WORLD_CLI_PARSER_HPP_
+#define _FASTDDS_HELLO_WORLD_CLI_PARSER_HPP_
+
+using eprosima::fastdds::dds::Log;
+
+class CLIParser
+{
+public:
+
+    CLIParser() = delete;
+
+    enum entity_kind
+    {
+        PUBLISHER,
+        SUBSCRIBER,
+        UNDEFINED
+    };
+
+    struct publisher_config
+    {
+        uint16_t samples = 0;
+    };
+
+    struct subscriber_config : public publisher_config
+    {
+        bool use_waitset = false;
+    };
+
+    struct hello_world_config
+    {
+        entity_kind entity = entity_kind::UNDEFINED;
+        publisher_config pub_config;
+        subscriber_config sub_config;
+    };
+
+    static void print_help(
+            uint8_t return_code)
+    {
+        std::cout << "Usage: hello_world <entity> [options]"                                    << std::endl;
+        std::cout << ""                                                                         << std::endl;
+        std::cout << "Entities:"                                                                << std::endl;
+        std::cout << "  publisher                       Run a publisher entity"                 << std::endl;
+        std::cout << "  subscriber                      Run a subscriber entity"                << std::endl;
+        std::cout << ""                                                                         << std::endl;
+        std::cout << "Common options:"                                                          << std::endl;
+        std::cout << "  -h, --help                      Print this help message"                << std::endl;
+        std::cout << "  -s <num>, --samples <num>       Number of samples to send or receive"   << std::endl;
+        std::cout << "                                  (Default: 0 [unlimited])"               << std::endl;
+        std::cout << "Subscriber options:"                                                      << std::endl;
+        std::cout << "  -w, --waitset                   Use waitset read condition"             << std::endl;
+        std::exit(return_code);
+    }
+
+    static hello_world_config parse_cli_options(
+            int argc,
+            char* argv[])
+    {
+        hello_world_config config;
+
+        std::string first_argument = argv[1];
+
+        if (first_argument == "publisher" )
+        {
+            config.entity = entity_kind::PUBLISHER;
+        }
+        else if ( first_argument == "subscriber")
+        {
+            config.entity = entity_kind::SUBSCRIBER;
+        }
+        else
+        {
+            if (first_argument == "-h" || first_argument == "--help")
+            {
+                print_help(EXIT_SUCCESS);
+            }
+            else
+            {
+                EPROSIMA_LOG_ERROR(CLI_PARSER, "parsing entity argument");
+                print_help(EXIT_FAILURE);
+            }
+        }
```

Comment:
```suggestion
```suggestion
        else if (first_argument == "-h" || first_argument == "--help")
        {
            print_help(EXIT_SUCCESS);
        }
        else
        {
            EPROSIMA_LOG_ERROR(CLI_PARSER, "parsing entity argument");
            print_help(EXIT_FAILURE);
        }
```
```